docs/dump_javadump.md

```diff
@@ -660,6 +678,206 @@ NULL           =================================
+3XMTHREADINFO3           Java callstack:
+4XESTACKTRACE                at java/lang/StringBuffer.ensureCapacityImpl(StringBuffer.java:696)
+4XESTACKTRACE                at java/lang/StringBuffer.append(StringBuffer.java:486(Compiled Code))
+5XESTACKTRACE                   (entered lock: java/lang/StringBuffer@0x957645B8, entry count: 1)
+4XESTACKTRACE                at java/lang/StringBuffer.append(StringBuffer.java:428(Compiled Code))
+4XESTACKTRACE                at HeapBreaker.main(HeapBreaker.java:34(Compiled Code))
+3XMTHREADINFO3           Native callstack:
+4XENATIVESTACK               (0xB6C535B3 [libj9prt29.so+0x3b5b3])
+4XENATIVESTACK               (0xB6C36F3E [libj9prt29.so+0x1ef3e])
+4XENATIVESTACK               (0xB6C5311E [libj9prt29.so+0x3b11e])
+4XENATIVESTACK               (0xB6C53216 [libj9prt29.so+0x3b216])
+4XENATIVESTACK               (0xB6C52CE3 [libj9prt29.so+0x3ace3])
+4XENATIVESTACK               (0xB6C36F3E [libj9prt29.so+0x1ef3e])
+4XENATIVESTACK               (0xB6C52C57 [libj9prt29.so+0x3ac57])
+4XENATIVESTACK               (0xB6C4E9CD [libj9prt29.so+0x369cd])
+4XENATIVESTACK               (0xB6C502FA [libj9prt29.so+0x382fa])
+```
+
+To simulate a Java `OutOfMemoryError`, our testcase works by repeatedly appending characters to a `StringBuffer` object in an infinite loop. The Java callstack shows the `HeapBreaker.main` method appending characters (`java/lang/StringGuffer.append`) until the method `java/lang/StringBuffer.ensureCapacityImpl()` throws the `OutOfMemoryError`.
+
+StringBuffer objects are wrappers for character arrays (`char[]`) and when the capacity of the underlying array is reached, the contents are automatically copied into a new, larger array. The new array is created in the `StringBuffer.ensureCapacity()` method, which more or less doubles the size of the old array. In our scenario, the array takes up all the remaining space in the Java heap.
+
+Although the Java code we used in this scenario was deliberately triggering an `OutOfMemoryError` in a spectacular way, similar allocation issues can and do occur when dealing with large data sets such as XML files.
+
+The next step in diagnosing the problems would be to open the system dump that gets generated automatically when an `OutOfMemoryError` occurs. Open
+the dump with the [Eclipse Memory Analyzer tool (MAT)](https://www.eclipse.org/mat/) and search for the `StringBuffer` object, which should provide further clues about what went wrong. A common
+example is seeing the same `String` duplicated over and over again, which might indicate that code is stuck in a loop.
+
+  <i class="fa fa-pencil-square-o" aria-hidden="true"></i> **Note:** If you want to use MAT to analyze your system dump, you must install the Diagnostic Tool Framework for Java (DTFJ) plugin in the Eclipse IDE. Select the following menu items:
+
+```
+Help > Install New Software > Work with "IBM Diagnostic Tool Framework for Java" >  
```

Comment:
Two concerns here:
1) OpenJ9 shouldn't redirect to IBM tools.  It would be better to direct to the Eclipse MAT downloads.
2) Have we confirmed that this works with OpenJ9?  I'd like confirmation that it works before this goes live

Reply:
I'll let @pshipton comment here. My understanding from earlier updates made in System dump was that the Eclipse MAT tool requires DTFJ to analyze the system dump. As this info came from Peter, my guess is that it has been tested.

Reply:
Thanks for the confirmation.  One more thing we should add to the todo list

Reply:
Opened an issue at OpenJ9 so this isn't lost: https://github.com/eclipse/openj9/issues/3164